src/OrleansServiceBus/Providers/Streams/EventHub/EventHubAdapterFactory.cs

```diff
@@ -283,6 +283,12 @@ private EventHubAdapterReceiver GetOrCreateReceiver(QueueId queueId)
             return cache;
         }
 
+        protected virtual IEventHubQueueCache CreateCache(IStreamQueueCheckpointer<string> checkpointer, 
```

Comment:
You can extend existing IEventHubQueueCache implementation by implementing a class inheriting EventHubQueueCache. EventHubQueueCache is an abstract class. Or does that not work for you? 
I think we are not debating extension through inheritance is better or through a factory pattern is better. It's if the factory fulfils your needs, imo we shouldn't introduce another pattern to bring more confusion.